lib/carthage_cache/configuration_validator.rb

```diff
@@ -69,11 +69,17 @@ def valid?
       validate.valid?
     end
 
+    def read_only?
+      (config.aws_access_key_id.nil? or config.aws_secret_access_key.nil?) and config.aws_profile.nil?
+    end
+
     def validate
       return missing_bucket_name unless has_bucket_name?
       return missing_aws_region unless has_aws_region?
-      return missing_aws_access_key_id unless has_aws_access_key_id?
-      return missing_aws_secret_access_key unless has_aws_secret_access_key?
+
+      return missing_aws_access_key_id if not has_aws_access_key_id? and has_aws_secret_access_key?
```

Comment:
TBH can't remember exactly 😄.
I guess this was for allowing to supply a profile instead of key+secret as env variables and if we can't find either profile or Key+Secret we treat it as a valid config but the script will run in read only mode (using wget to download the file)